content/blog/new-graphic-features/index.Rmd

```diff
@@ -0,0 +1,301 @@
+
+### The graphics engine
+
+What unites base and grid graphics and sets them apart from e.g. rgl is that they both call into the same low level C API provided by R, the **graphics engine**. The graphics engine is responsible for communicating with the graphics devices while also providing a selected range of utility functionality common for both base and grid graphics. It is because of this abstraction that graphics in R is largely decoupled from how it is outputted, be it on screen, in a file, or directly to a printer.
+
+While it sounds nice and neat when it is all laid out like this, the current structure and division has grown out over many years and the boundary between the graphic systems, the graphics engine, and the graphic devices are blurry. Still, the design is much more mature than what we see in other languages and as graphics/data-viz developers in R we are pretty spoiled compared to our peers in other languages --- perhaps without really knowing it.
+
+## A fragmented future
+
+With the division of responsibility described above, there are many points in the pipeline that may impose limitations in functionality. The graphic system might not provide the higher-level API to use functionality in the graphics engine, or a graphic device might not provide support for instructions coming in from the graphics engine. While this has mainly been a hypothetical situation prior to R 4.1, it is now the new reality. The new features in the graphics engine was implemented along with high-level support in grid, and low level support in the pdf device along with the cairo based devices. This leaves base plot in the dark and also exclude a range of built-in devices, including the default devices on Windows and macOS. At this point, where high level support from e.g. ggplot2 is still not present, it might not be a big problem as you will probably use these features quite deliberately and know their limitations in support. In the future, however, this could lead to surprises.
+
+As users, this fragmentation is most apparent in the choice of graphic device. After all, you don't expect the graphic system to be capable of something outside of its API, simply because new features were announced for the graphics device. However, if a graphic device lack support it will simply not use the new features and you may end up surprised at what it renders.
```

Comment:
```suggestion
As users, this fragmentation is most apparent in the choice of graphic device. After all, you don't expect the graphic system to be capable of something outside of its API, simply because new features were announced for the graphics device. However, if a graphic device lacks support it will simply not use the new features, and you may end up surprised at what it renders.
```